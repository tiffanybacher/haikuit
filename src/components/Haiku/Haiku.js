import React from 'react';

const Haiku = (props) => {
  let { title, line1, line2, line3 } = props;

  if (!props.id) {
    title = 'A Haiku about no haikus';
    line1 = 'You have no haikus!';
    line2 = 'Oh no! How sad. But don\'t fear!';
    line3 = 'Write one or a few!';
  }

  return (
    <article className={!props.id ? "no-hover Haiku" : "Haiku"}>
      <p className="title"><b>{title}</b></p>
      <p className="line-1">{line1}</p>
      <p className="line-2">{line2}</p>
      <p className="line-3">{line3}</p>
    </article>
  );
}

export default Haiku;