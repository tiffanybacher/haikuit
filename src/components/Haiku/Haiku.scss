.Haiku {
  background: white;
  line-height: 1.3em;
  margin: 0 0 20px 0;
  padding: 20px;
  text-align: center;
}

.title {
  margin: 0 0 5px 0;
}

@media only screen and (min-width: 768px) {
  .Haiku {
    font-size: 1.1em;
    padding: 20px 50px;
  }
}

@media only screen and (min-width: 1000px) {
  .Haiku {
    font-size: 1.2em;
    padding: 20px 75px;
  }
}

@media only screen and (min-width: 1200px) {
  .Haiku {
    padding: 20px 250px;
  }
}
