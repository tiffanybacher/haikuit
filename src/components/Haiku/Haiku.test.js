import React from 'react';
import { shallow } from 'enzyme';
import Haiku from './Haiku';

describe('Haiku', () => {
  let mockHaiku;
  let wrapper;

  beforeEach(() => {
    mockHaiku = { title: 'title', id: 123 }
    wrapper = shallow(<Haiku />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if haiku is passed in', () => {
    wrapper = shallow(<Haiku { ...mockHaiku } />);

    expect(wrapper).toMatchSnapshot();
  });
});