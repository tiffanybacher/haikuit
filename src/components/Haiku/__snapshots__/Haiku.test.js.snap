// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Haiku should match snapshot 1`] = `
<article
  className="no-hover Haiku"
>
  <p
    className="title"
  >
    <b>
      A Haiku about no haikus
    </b>
  </p>
  <p
    className="line-1"
  >
    You have no haikus!
  </p>
  <p
    className="line-2"
  >
    Oh no! How sad. But don't fear!
  </p>
  <p
    className="line-3"
  >
    Write one or a few!
  </p>
</article>
`;

exports[`Haiku should match snapshot if haiku is passed in 1`] = `
<article
  className="Haiku"
>
  <p
    className="title"
  >
    <b>
      title
    </b>
  </p>
  <p
    className="line-1"
  />
  <p
    className="line-2"
  />
  <p
    className="line-3"
  />
</article>
`;
